services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java

```diff
@@ -0,0 +1,753 @@
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.cloudfront;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.SecureRandom;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.spec.InvalidKeySpecException;
+import java.time.Instant;
+import java.util.Base64;
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.annotations.ThreadSafe;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.services.cloudfront.internal.auth.Pem;
+import software.amazon.awssdk.services.cloudfront.internal.auth.Rsa;
+import software.amazon.awssdk.utils.IoUtils;
+import software.amazon.awssdk.utils.StringUtils;
+
+@Immutable
+@ThreadSafe
+@SdkPublicApi
+public final class CloudFrontUtilities {
+
+    private CloudFrontUtilities() {
+    }
+
+    /**
+     * Returns a signed URL with a canned policy that grants universal access to
+     * private content until a given date.
+     * For more information, see <a href=
+     * "https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-creating-signed-url-canned-policy.html"
+     * >Creating a signed URL using a canned policy</a>.
+     *
+     * @param protocol
+     *            The protocol of the URL - HTTP or HTTPS
+     * @param distributionDomain
+     *            The domain name of the distribution
+     * @param s3ObjectKey
+     *            The s3 key of the object
+     * @param privateKeyFile
+     *            The private key file. RSA private key (.pem) and pkcs8 (.der)
+     *            files are supported.
+     * @param keyPairId
+     *            Identifier of a public/private certificate keypair already
+     *            configured in your Amazon Web Services account.
+     * @param expirationDate
+     *            The expiration date of the signed URL in UTC
+     * @return The signed URL.
+     */
+    public static String getSignedUrlWithCannedPolicy(Protocol protocol,
+                                                      String distributionDomain,
+                                                      String s3ObjectKey,
+                                                      File privateKeyFile,
+                                                      String keyPairId,
+                                                      Instant expirationDate) throws InvalidKeySpecException, IOException {
```

Comment:
Create a POJO request class to represent the input parameters. You can put it under `/model`

Reply:
Added CloudFrontSignerRequest under /model directory